src/trezor-keyring.ts

```diff
@@ -420,17 +493,20 @@ class TrezorKeyring extends EventEmitter {
     await wait(status === 'just unlocked' ? DELAY_BETWEEN_POPUPS : 0);
 
     const response = await TrezorConnect.ethereumSignTypedData({
-      path: this._pathFromAddress(address),
+      path: this.#pathFromAddress(address),
       data: {
-        types: { EIP712Domain, ...otherTypes },
+        types: { EIP712Domain, ...otherTypes } as T,
```

Comment:
Forgive me if this has been asked already, but is there any way that we can avoid this typecasting?

Reply:
@mcmire TS complains that the object might be instantiated with a different subset of `MessageTypes`, as we destructure `dataWithHashes.types` to give default values.
We can maybe avoid the cast and assign default values anyway in this way:
```typescript
const {
      types,
      message = {},
      domain = {},
      primaryType,
      domain_separator_hash,
      message_hash,
    } = dataWithHashes;
    
// ...

data: {
    types: { ...types, EIP712Domain: types.EIP712Domain ?? [] },
```

What do you think?

Reply:
Yeah I think that would make sense!